--- a/libusb/os/darwin_usb.h	2023-03-19 12:07:53
+++ b/libusb/os/darwin_usb.h	2023-03-19 12:07:47
@@ -165,7 +165,8 @@
   #define __has_builtin(x) 0  // Compatibility with non-clang compilers.
 #endif
 #if __has_builtin(__builtin_available)
-  #define HAS_CAPTURE_DEVICE() __builtin_available(macOS 10.10, *)
+//  #define HAS_CAPTURE_DEVICE() __builtin_available(macOS 10.10, *)
+  #define HAS_CAPTURE_DEVICE() 0
 #else
   #define HAS_CAPTURE_DEVICE() 0
 #endif
